### PR TITLE
Assignment 9

### PR DESCRIPTION
**Pull request by TatyanaHubbard**

### CODE REVIEW

File: ShoppingList/ShoppingList/ShoppingList.csproj [hunk 1]
```diff
@@ -6,4 +6,15 @@
     <UseWPF>true</UseWPF>
```
Comment (mmwoodfo) [inline ShoppingList/ShoppingList/ShoppingList.csproj]:
```suggestion
    <UseWPF>true</UseWPF>
    <NoWarn>AD0001</NoWarn>
```
You seem to have warnings set to be treated errors - I believe you can ignore this warning & your program should run. Alternatively, you can set treating warnings as errors to false.

Comment (TatyanaHubbard) [reply ShoppingList/ShoppingList/ShoppingList.csproj]: Thank you! I wasn't sure if we were supposed to or not

**Comment by dbosco051394** [comment]
On line 31 of your XAML        
<Button Content="Add Item" Command="{Binding AddItem}" CommandParameter="{Binding Text, ElementName=NewItem}" Grid.Row="2" Background="LightSlateGray"/>

You have ElementName set to 'NewItem' but there is no context for NewItem in your view model. Changing this to 'NewName' could help your issue with adding not working potentially.

File: ShoppingList/ShoppingList/MainWindowViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        public RelayCommand? AddItem { get; }
+
+        public MainWindowViewModel()
+        {
+            AddItem = new RelayCommand(OnAddItem);
+            _NewName = "";
+        }
+
+        private void OnAddItem()
```
Comment (matHopg) [inline ShoppingList/ShoppingList/MainWindowViewModel.cs]: Your add item seems to only add items to the code behind but does not seem to update the list in your xaml am I mistaken here?

File: ShoppingList/ShoppingList/MainWindow.xaml [hunk 3]
```diff
@@ -4,9 +4,62 @@
...
+        
+        <ListBox x:Name="ShoppingListBox" 
+                 Grid.Row="0"
+                 Background="WhiteSmoke"
+                 ItemsSource="{Binding MyShoppingList}" 
+                 SelectedItem="{Binding SelectedShopItem}">
+            
+            <ListBox.ItemTemplate>
+                <DataTemplate DataType="{x:Type local:ShopItem}">
+                    <StackPanel>
```
Comment (Keboo) [inline ShoppingList/ShoppingList/MainWindow.xaml]: `StackPanel` is not adding anything here. You can just have the TextBlock be there directly.

File: ShoppingList/ShoppingList/ShoppingList.csproj [hunk 4]
```diff
@@ -4,6 +4,21 @@
     <OutputType>WinExe</OutputType>
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <UseWPF>true</UseWPF>
+    <Nullable>enable</Nullable>
+    <NoWarn>AD0001</NoWarn>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (Keboo) [inline ShoppingList/ShoppingList/ShoppingList.csproj]: In general you don't want these in a `PropertyGroup` with a condition, and instead just put them inside of the default property group above. This type of thing usually happens when you turn these on from the project properties page inside of VS.

File: ShoppingList/ShoppingList/VisibilityConverter.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+using System.Windows.Data;
+namespace ShoppingList
+{
+    public class VisibilityConverter : IValueConverter
+    {
+        public object Convert(object value, Type? targetType = null, object? parameter = null, CultureInfo? culture = null)
+        {
+            if (value is ShopItem)
+            {
+                return "Visible";
```
Comment (Keboo) [inline ShoppingList/ShoppingList/VisibilityConverter.cs]: There is a `Visibility` enum that would have been better here. The default WPF type converters are the reason why passing a string works. The actual Visibility property uses the enum.

**Pull request closed without merging**